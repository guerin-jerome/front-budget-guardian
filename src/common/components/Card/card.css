.card {
  margin-top: 20px;
  border: 1px solid #ddd;
  padding: 12px;
}
