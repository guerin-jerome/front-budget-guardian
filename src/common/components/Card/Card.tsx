import { CardProps } from "./card.type";
import classNames from "classnames";
import "./card.css";

export const Card = ({ children, className }: CardProps) => (
  <div className={classNames("card", className)}>{children}</div>
);
