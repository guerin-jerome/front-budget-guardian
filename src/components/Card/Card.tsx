import { TWithChildren } from "../components.type";
import "./card.css";

export const Card = ({ children }: TWithChildren) => (
  <div className="card">{children}</div>
);
