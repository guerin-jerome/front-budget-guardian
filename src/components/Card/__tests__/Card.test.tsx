import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { Card } from "../Card";

describe("<Card />", () => {
  it("matchs snapshot", () => {
    const { container } = render(<Card>Hello</Card>);
    expect(container).toMatchSnapshot();
  });

  it("renders well", () => {
    render(<Card>Hello</Card>);
    const cardText = screen.getByText("Hello");
    expect(cardText).toBeDefined();
  });
});
