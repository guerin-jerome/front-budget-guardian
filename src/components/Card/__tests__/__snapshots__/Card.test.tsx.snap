// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Card /> > matchs snapshot 1`] = `
<div>
  <div
    class="card"
  >
    Hello
  </div>
</div>
`;
